packages/dart/noports_core/lib/src/sshnpd/sshnpd_config.dart

```diff
@@ -0,0 +1,356 @@
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:config/config.dart';
+import 'package:noports_core/sshnp_foundation.dart';
+import 'package:yaml/yaml.dart';
+
+const OptionGroup mainGroup = OptionGroup("atSign Options");
+const OptionGroup deviceConfigGroup = OptionGroup(
+  "Device Configuration Options",
+);
+const OptionGroup accessControlGroup = OptionGroup("Access Control Options");
+const OptionGroup sshGroup = OptionGroup("Sshd Configuration Options");
+const OptionGroup runtimeGroup = OptionGroup("Runtime Options");
+
+File _defaultConfigFilePath() {
+  switch (Platform.operatingSystem) {
+    case "windows":
+      final programData = Platform.environment['ProgramData'];
+      return File("$programData/NoPorts/sshnpd.yaml");
+    case "macos":
+      return File("/Library/Application Support/NoPorts/sshnpd.yaml");
+    default:
+      return File("/etc/noports/sshnpd.yaml");
+  }
+}
+
+enum SshnpdOption<V> implements OptionDefinition<V> {
```

Comment:
This enum represents how configuration is parsed, it is a wrapper around package:args which also supports config file parsing and environment variable parsing.

You will see that there are prefixes of `arg`, `env`, and `config` where appropriate.